fuel/app/classes/materia/widget/asset/storage/s3.php

```diff
@@ -20,14 +20,117 @@ public static function instance(array $config): Widget_Asset_Storage_Driver
 	}
 
 	/**
-	 * Create a lock on a specific size of an asset.
+	 * Create a lock on a specific size of an asset for a period of time
+	 * Used to prevent multiple requests from using excessive resources.
+	 * For object locking to work, the bucket must have versioning enabled
+	 * @param  string $id   Asset Id to lock
+	 * @param  string $size Size of asset data to lock
+	 */
+	public function lock_for_period(string $id, string $size): void
+	{
+		$s3 = $this->get_s3_client();
+
+		try {
+			$s3->putObjectRetention([
+				'Bucket' => static::$_config['bucket'],
+				'Key' => $this->get_key_name($id, $size),
+				'Retention' => [
+					'Mode' => 'GOVERNANCE',
+					'RetainUntilDate' => new \DateTime('+1 hour'),
+				],
+				// 'VersionId' => '<string>',
+				// 'BypassGovernanceRetention' => true || false,
+				// 'ChecksumAlgorithm' => 'CRC32|CRC32C|SHA1|SHA256',
+				// 'ContentMD5' => '<string>',
+				// 'ExpectedBucketOwner' => '<string>',
+				// 'RequestPayer' => 'requester',
```

Comment:
Are these values still needed?

Reply:
No. I left them commented just in case future developers wanted to know the other possible parameters in this version of s3, but I suppose they could just look at the docs.